### PR TITLE
Improve Elyses Transformative Enchantments

### LINKED ISSUE #1326: Improve Concept Exercise: Elyses Transformative Enchantments (array-transformations)
The goal of this task is to make sure the Concept Exercise "Elyses Transformative Enchantments" conform to the latest specification. As part of that, the concept `array-transformations` has to be created.

The docs can be found here
- [What are Concept Exercises?](https://exercism.org/docs/building/product/concept-exercises)
- [Concept Exercises](https://exercism.org/docs/building/tracks/concept-exercises)
- [Concepts](https://exercism.org/docs/building/tracks/concepts)

Additionally you can look at the `bird-watcher` concept exercise for reference.

The improvement should include changing the introduction.md file of the exercise to contain some minimal information so that the student is able to solve the exercise. For now it is ok to use that same content as introduction.md and about.md for the Concept as well.

Also ensure that
* there are good blurbs for the concept and exercise
* the tests are declarative (see https://github.com/exercism/javascript/issues/1019 for an example)
* each test case is "self-contained" so that the student sees meaningful output from the test runner (no forEach, no helper functions defined in other parts of the test file etc.)

Here is a general idea for the introduction that was discussed before:
Show how to do something like map or filter with a simple for loop, then show how to do it with the helper method instead. Afterwards provide a table with the common helpers. Just the method name as a link (e.g. to MDN) and a short explanation what this is for. We think its ok if the student reads up on some details of these methods on their own, we don't want to show and explain every single one in detail. Also the list can contain more things, than needed for the exercise (we count that info as "necessary to understand the concept" in contrast to "necessary to solve the exercise").

It might make sense to replace the `fill` task of the exercise (the last one) with `reverse` instead as that is probably more commonly used.

### PR DESCRIPTION
**Pull request by junedev**
Resolves #1326 (the remaining todos)

Note that the exercise might have some ambiguities currently to whether the original array can be modified or not. For some cases, like when we expect the student to use `map`, I added a test case to clarify the expected behavior. For other cases it is left up to the student for now. Let's see how this is received first. If a lot of students are confused, we can still add more clarification later.

### CODE REVIEW

File: exercises/concept/elyses-transformative-enchantments/.docs/introduction.md [hunk 1]
```diff
@@ -1,23 +1,181 @@
 # Introduction
 
-In JavaScript, the `Array` class has many powerful built-in functions for transforming arrays. These functions make it much easier to do things than it otherwise would be using a simple for loop or more direct manipulation.
+In JavaScript, the `Array` class has many powerful built-in methods for transforming arrays.
+These methods make it much easier to convert one array into another than it otherwise would be using a simple for loop or more direct manipulation.
 
-Example of transformation using a for loop and second array:
+Some methods are [_pure_][pure-function-definition], meaning that they no not modify the original array.
+They return a new one instead.
+Other methods however manipulate the array they are called on and not **not** return the modified array.
```
Comment (pertrai1) [inline exercises/concept/elyses-transformative-enchantments/.docs/introduction.md:8]: Should this have **not** listed twice?

File: exercises/concept/elyses-transformative-enchantments/.docs/introduction.md [hunk 2]
```diff
@@ -1,23 +1,181 @@
...
+
+Here is a presentation of some of the most commonly used methods.
```
Comment (pertrai1) [inline exercises/concept/elyses-transformative-enchantments/.docs/introduction.md:10]: Should there be a link to the presentation

File: exercises/concept/elyses-transformative-enchantments/.docs/introduction.md [hunk 3]
```diff
@@ -1,23 +1,181 @@
...
+
+```javascript
+const arr = ['1', '2', '5', '6'];
+
+// Insert an element at index 2
+arr.splice(2, 0, '3');
+console.log(arr);
+// => ['1', '2', '3', '5', '6']
+
+// Remove 2 elements, starting at index 3 an insert 2 elements
```
Comment (pertrai1) [inline exercises/concept/elyses-transformative-enchantments/.docs/introduction.md:120]: I think **an** should be **and**

File: exercises/concept/elyses-transformative-enchantments/.docs/introduction.md (hunk 2 above)
Comment (junedev) [reply exercises/concept/elyses-transformative-enchantments/.docs/introduction.md:10]: What is meant here is that the methods are presented below. I rephrased the sentence to make this more clear.

File: exercises/concept/elyses-transformative-enchantments/.docs/introduction.md (hunk 1 above)
Comment (junedev) [reply exercises/concept/elyses-transformative-enchantments/.docs/introduction.md:8]: "not not" should be "do not"

**Comment by junedev** [comment]
@pertrai1 Thanks for the review! I fixed the things you mentioned in all the places.

**Pull request merged**